### PR TITLE
Switch to Docker build

### PR DESCRIPTION
**Pull request by nadade**
- updated Dredd contributiing URL
- `./docker/build.sh` to build an image with gems
- `./docker/serve.sh` to run jekyll at localhost:4000
- moved deploy instructions to `./deploy.sh`
- updated CircleCI build process to use Docker

### CODE REVIEW

File: _posts/2016-08-26-Test-Your-Swagger-API.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Finally, I can't even express here how awesome it is to work on a project like D
 [hooks]: http://dredd.readthedocs.io/en/latest/hooks/
 [dredd-example]: https://github.com/apiaryio/dredd-example/
 [how-to-guides]: http://dredd.readthedocs.io/en/latest/how-to-guides/
-[contributing-guidelines]: https://github.com/apiaryio/dredd/blob/master/CONTRIBUTING.md
+[contributing-guidelines]: https://dredd.readthedocs.io/en/latest/contributing/
```
Comment (abtris) [inline _posts/2016-08-26-Test-Your-Swagger-API.md]: I'm not sure if good this change add to docker PR.

**Comment by abtris** [comment]
👍

File: _posts/2016-08-26-Test-Your-Swagger-API.md (hunk 1 above)
Comment (nadade) [inline _posts/2016-08-26-Test-Your-Swagger-API.md]: @abtris I can issue a separate PR for the link if you'd like, felt like an overhead (the old link doesn't exist anymore, found out while testing)

Comment (JackuB) [inline _posts/2016-08-26-Test-Your-Swagger-API.md]: Keep it in pls, CI won't pass without it anyway and it's a good practice to keep those links checked IMHO

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ENV REFRESHED_AT 2016-10-14
+
+ADD ./Gemfile /srv/jekyll/Gemfile
+ADD ./Gemfile.lock /srv/jekyll/Gemfile.lock
+
+WORKDIR /srv/jekyll
+RUN chown -R jekyll /srv/jekyll
+RUN bundle
+USER jekyll
+ENV LC_ALL C.UTF-8
```
Comment (JackuB) [inline Dockerfile]: Reading about LC_ALL made me sick

File: _posts/2016-08-26-Test-Your-Swagger-API.md (hunk 1 above)
Comment (abtris) [inline _posts/2016-08-26-Test-Your-Swagger-API.md]: ok

**Pull request merged**